x-pack/plugins/maps/public/classes/styles/vector/properties/dynamic_style_property.tsx

```diff
@@ -345,4 +355,44 @@ export class DynamicStyleProperty<T>
       />
     );
   }
+
+  getMbPropertyName() {
+    if (!this._field) {
+      return '';
+    }
+
+    let targetName;
+    if (this.supportsMbFeatureState()) {
+      // Base case for any properties that can support feature-state (e.g. color, size, ...)
+      // They just re-use the original property-name
+      targetName = this._field.getName();
+    } else {
+      if (this._field.canReadFromGeoJson()) {
```

Comment:
Lets simplify this `if` to the below. There is no need for additional conditional operator inside if statement.

```
    if (this._field.canReadFromGeoJson() && this._field.supportsAutoDomain()) {
        // Geojson-sources can support rewrite
        // e.g. field-formatters will create duplicate field
        targetName = getComputedFieldName(this.getStyleName(), this._field.getName());
      } else {
        // Non-geojson sources (e.g. mvt)
        targetName = this._field.getName();
      }
```